src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,1515 @@
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* Check if ark_mem was allocated */
+  if (ark_mem->MallocDone == SUNFALSE)
+  {
+    arkProcessError(ark_mem, ARK_NO_MALLOC, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NO_MALLOC);
+    return (ARK_NO_MALLOC);
+  }
+
+  /* Check that fe is supplied */
+  if (fe == NULL)
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NULL_F);
+    return (ARK_ILL_INPUT);
+  }
+
+  /* Check that fi is NULL until IMEX module is ready */
+  if (fi != NULL)
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "\n\nNO IMEX-LSRK support yet, set fi = NULL");
+    return (ARK_ILL_INPUT);
+  }
+
+  /* Check for legal input parameters */
+  if (y0 == NULL)
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NULL_Y0);
+    return (ARK_ILL_INPUT);
+  }
+
+  /* Copy the input parameters into ARKODE state */
+  step_mem->fe = fe;
+  step_mem->fi = fi;
+
+  /* Initialize main ARKODE infrastructure */
+  retval = arkInit(arkode_mem, t0, y0, FIRST_INIT);
+  if (retval != ARK_SUCCESS)
+  {
+    arkProcessError(ark_mem, retval, __LINE__, __func__, __FILE__,
+                    "Unable to initialize main ARKODE infrastructure");
+    return (retval);
+  }
+
+  /* Initialize all the counters, flags and stats */
+  step_mem->nfe            = 0;
+  step_mem->nfi            = 0;
+  step_mem->domeignfe      = 0;
+  step_mem->ndomeigupdates = 0;
+  step_mem->stagemax       = 0;
+  step_mem->sprmax         = 0;
+  step_mem->sprmin         = 0;
+  step_mem->nstsig         = 0;
+  step_mem->newdomeig      = SUNTRUE;
+  step_mem->jacatt         = SUNFALSE;
+
+  return (ARK_SUCCESS);
+}
+
+/*===============================================================
+  Interface routines supplied to ARKODE
+  ===============================================================*/
+
+/*---------------------------------------------------------------
+  lsrkStep_Resize:
+
+  This routine resizes the memory within the LSRKStep module.
+  ---------------------------------------------------------------*/
+int lsrkStep_Resize(ARKodeMem ark_mem, N_Vector y0,
+                    SUNDIALS_MAYBE_UNUSED sunrealtype hscale,
+                    SUNDIALS_MAYBE_UNUSED sunrealtype t0, ARKVecResizeFn resize,
+                    void* resize_data)
+{
+  ARKodeLSRKStepMem step_mem;
+  sunindextype lrw1, liw1, lrw_diff, liw_diff;
+  int retval;
+
+  /* access ARKodeLSRKStepMem structure */
+  retval = lsrkStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* Determine change in vector sizes */
+  lrw1 = liw1 = 0;
+  if (y0->ops->nvspace != NULL) { N_VSpace(y0, &lrw1, &liw1); }
```

Comment:
`lsrkStep_CheckNVector` already checks this exists

Reply:
I couldn't quite get that. `***Step_Resize` functions are pretty much the same for all of the other modules. Can you elaborate on what you mean?

Reply:
I meant
```suggestion
  N_VSpace(y0, &lrw1, &liw1);
```
If the other modules have it, we can leave it for consistency.